airbyte-commons/src/main/java/io/airbyte/commons/json/JsonSchemas.java

```diff
@@ -220,6 +221,9 @@ public static List<String> getType(final JsonNode jsonNode) {
         return List.of(jsonNode.get(JSON_SCHEMA_TYPE_KEY).asText());
       }
     }
+    if (jsonNode.has(ENUM_TYPE)) {
+      return List.of(ENUM_TYPE);
```

Comment:
Do we need to return a new type? Can we say the enum type is always a string instead of adding a new type? i.e. let's return `List.of(STRING_TYPE)`. Does JSONSchema say anything about this? I thought the enum field was a modifier for string types only but I could be wrong about what the specification says.

Also, is just having the enum field considered valid JSONSchema? Or is it just something that _we_ do? If the latter, let's add a comment on line 224 calling that out so that it is clear. I'm fine with us being a little more permissive, but I want to be clear about whether we are breaking the normal convention or not.

In the future, it would be good to add to the standard tests for a connector a validation that the specs it uses match the subset of JSONSchema functionality that we support (certainly out of scope for this issue).

Reply:
created this issue for the future work i proposed above: https://github.com/airbytehq/airbyte/issues/12330

Reply:
I have seen a stack-overflow example with integer type for the enum while searching for the best apporach to this fix https://stackoverflow.com/questions/30924271/correct-way-to-define-array-of-enums-in-json-schema. So I wanted to introduce specific type because nothing tell me in the specs that a given type need to be enforce.

Other source don't explicit the type https://json-schema.org/understanding-json-schema/reference/generic.html#enumerated-values

Reply:
> [6.1.2.](https://json-schema.org/draft/2020-12/json-schema-validation.html#rfc.section.6.1.2) [enum](https://json-schema.org/draft/2020-12/json-schema-validation.html#enum)
> The value of this keyword MUST be an array. This array SHOULD have at least one element. Elements in the array SHOULD be unique.
> 
> An instance validates successfully against this keyword if its value is equal to one of the elements in this keyword's array value.
> 
> Elements in the array might be of any type, including null.
-- https://json-schema.org/draft/2020-12/json-schema-validation.html#rfc.section.6.1.2

huh. you are right. enums are not restricted to strings.

I think the key thing here is that `enum` is _not_ a JSONSchema type. It is a modifier field. 
> [6.1.1.](https://json-schema.org/draft/2020-12/json-schema-validation.html#rfc.section.6.1.1) type
> The value of this keyword MUST be either a string or an array. If it is an array, elements of the array MUST be strings and MUST be unique.
>
> String values MUST be one of the six primitive types ("null", "boolean", "object", "array", "number", or "string"), or "integer" which matches any number with a zero fractional part.
>
> An instance validates if and only if the instance is in any of the sets listed for this keyword.

-- https://json-schema.org/draft/2020-12/json-schema-validation.html#rfc.section.6.1.1

My reading of this is that it is invalid for a JSONSchema object to not have a `type` field set. I think right now we permit it and assume if it is missing it is an `object`. That permissiveness bites us now, because really the assumption that would make sense if no type is set but `enum` is set is probably string, but this starts getting complicated.

I don't think we should introduce `enum` as a new type. It isn't real JSONSchema, but someone who is not very familiar with is not going to know that. I think in the short term I would suggest: 1) if enum but type not set, assume string. 2) fix the invalid specs (though this may be very hard to do).

Reply:
Ha! I'm not crazy. Other people are confused by this too: https://github.com/json-schema/json-schema/issues/172. But also means my previous statement is wrong. If `type` is not set, you can make no assumptions about the type of the object. It can be anything!

Okay, based on that I do think my suggestions from the previous section holds. I think for `specs`, _we_ should not allow the `type` to not be set. There is no reason to saddle ourselves with that ambiguity. That said, there is not a "free" path to that without giving connector developers a tool to validate their schemas. Thus, a shim for now that just assumes that if `type` is not set and `enum` is set that we are dealing with a string seems like an okay hack. And then we can remove it once we provide adequate tooling for connector developers in this future issue https://github.com/airbytehq/airbyte/issues/12330. WDYT?

Reply:
I think it is the right way to go, even if the types are not specify. In this case we want the same behavior for string and enum since both are non-traversable and thus are a stop condition in the recursion.

We can then work on fixing the schema.